### ISSUE TITLE
build and generate package fail

### ISSUE THREAD

**Issue by Varorbc** [dotnet/sdk#3553]
**Version Used**: 3.0.100-preview8-013656

###  Question 1

**Steps to Reproduce**:

1. Directory.Build.props

```csharp
    <Project>
      <PropertyGroup>
        <TargetFramework>net472</TargetFramework>
      </PropertyGroup>
    </Project>
```

2. ClassLibrary1.csproj

```csharp
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFrameworks>net452</TargetFrameworks>
    <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
  </PropertyGroup>
  
</Project>
```

3. dotnet build ClassLibrary1.csproj

**Expected Behavior**: build successful

**Actual Behavior**:`C:\Program Files\dotnet\sdk\3.0.100-preview8-013656\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(234,5): error NETSDK1005: 资产文件“C:\Users\PC\Desktop\ClassLibrary1\ClassLibrary1\obj\project.assets.json”没有“.NETFramework,Version=v4.7.2”的目标。确保已运行还原，且“net472”已包含在项目的 TargetFrameworks 中。 [C:\Users\PC\Desktop\ClassLibrary1\ClassLibrary1\ClassLibrary1.csproj]`

PS: build by Visual Studio 16.2.2 is successful

###  Question 2

**Steps to Reproduce**:

1. Directory.Build.props

```csharp
    <Project>
      <PropertyGroup>
        <TargetFramework>net472</TargetFramework>
      </PropertyGroup>
    </Project>
```

2. ClassLibrary1.csproj

```csharp
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFrameworks>net452;net472</TargetFrameworks>
    <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
  </PropertyGroup>
  
</Project>
```

3. dotnet build ClassLibrary1.csproj

**Expected Behavior**: generate package successful

**Actual Behavior**:build successful but no generate package

PS: build and generate package  by Visual Studio 16.2.2 is successful

**Comment by Varorbc** [comment]
@dsplaisted may you fix these two problems?

**Comment by dasMulli** [comment]
I think this is mostly an issue with NuGet.

I've been telling people not to use `TargetFramework` or `TargetFrameworks` inside `Directory.Build.props` files since they interfere with a lot of stuff - from VS' project system selection, conditional imports generated by NuGet to restore behaviour.

Setting both the singular and plural versions confuse targets about if it is a single-TFM project or a multi-targeting project in many places.

Scenario 1 should also happen without `GeneratePackageOnBuild` - The restore graph generation logic only respects the plural version if present. The singular version is always expected to reference one of the TFMs inside the plural version.

Scenario 2: GeneratePackageOnBuild does nothing since it thinks it's inside an inner build - it tests the `IsInnerBuild` property which is set by the pack targets as
```xml
<IsInnerBuild Condition="'$(TargetFramework)' != '' AND '$(TargetFrameworks)' != ''">true</IsInnerBuild>
```

When calling `dotnet pack` on the same project, it would just complain about not being able to find the built dll for the other TFM specified.

e.g.
error NU5026: The file '…/bin/Debug/net452/testlib.dll' to be packed was not found on disk.

TL;DR setting `TargetFramework` AND `TargetFrameworks` in a project file is dangerous, more so in `Directory.Build.props´ files.
